doc/spec.md

```diff
@@ -2348,7 +2355,62 @@ foo({ id: 1234, name: false });    // Error, name of wrong type
 foo({ name: "hello" });            // Error, id required but missing
 ```
 
-### <a name="3.11.5"/>3.11.5 Excess Properties
+### <a name="3.11.5"/>3.11.5 Comparability
+
+*TODO: Document the base primitive type.
+
+*Types are required to be *comparable* in certain circumstances, such as part of when checking whether two values of given types might be equal at runtime using operators like '===', or when using a type assertion.
+
+*S* is ***comparable to*** a type *T*, and *T* is ***comparable from*** *S*, if one of the following is true:
+
+* *S* and *T* are identical types.
+* *S* or *T* is the Any type.
+* *S* or *T* is an enum type and the other is the primitive type Number.
+* *S* is a literal type and *T* is the base primitive type of S.
+* *S* is a union type and some constituent type of *S* is comparable to *T*.
```

Comment:
TODO: Excess property check goes above this line.

Reply:
I don't believe that's relevant to comparability.